src/components/Counter/Counter.jsx

```diff
@@ -1,41 +1,65 @@
-import { useEffect, useState } from 'react'
+import { useEffect, useState, useReducer } from 'react'
 
+const intitalValue = { currentCount: 0 }
 const pinkRGB = `rgb(236, 72, 153)`
 
+function countReducer(count, action) {
+  switch (action.type) {
+    case 'incremented': {
+      return { currentCount: count.currentCount + 1 }
+    }
+    case 'decremented': {
+      return { currentCount: count.currentCount - 1 }
+    }
+    case 'reset': {
+      return { currentCount: 0 }
+    }
+    default: {
+      throw Error(`Uknown action: ${action.type}`)
+    }
+  }
+}
+
 export default function Counter() {
-  const [count, setCount] = useState(0)
+  const [count, dispatch] = useReducer(countReducer, intitalValue)
   const [currentColor, setCurrentColor] = useState(pinkRGB)
```

Comment:
You can also add your color to the reducer if you want to handle both states in one place